src/main/scala/io/citrine/lolo/PredictionResult.scala

```diff
@@ -44,6 +47,61 @@ trait PredictionResult[+T] {
   def getGradient(): Option[Seq[Vector[Double]]] = None
 }
 
+/**
+ * Additional regression-specific interface
+ */
+trait RegressionResult extends PredictionResult[Double] {
+  /**
+   * Get the 1-\sigma Prediction Interval for each prediction, if possible
+   *
+   * https://en.wikipedia.org/wiki/Prediction_interval
+   * The prediction interval is typically the sum of the (square) bias, variance, and irreducible error
+   */
+  def getPredictionInterval(): Option[Seq[Double]] = None
+
+  /**
+   * Get the 1-\sigma Confidence Interval for each prediction, if possible
+   * https://en.wikipedia.org/wiki/Confidence_interval
+   */
+  def getConfidenceInterval(): Option[Seq[Double]] = None
+
+  /**
+   * Get the estimated magnitude of the bias of each prediction, if possible
+   *
+   * See https://en.wikipedia.org/wiki/Bias%E2%80%93variance_tradeoff
+   */
+  def getBias(): Option[Seq[Double]] = None
+
+  /**
+   * Get the estimated magnitude of the sqrt of the variance of each prediction, if possible
+   *
+   * The sqrt is so that the bias, variance, and irreducible error have the same scale.
+   * See https://en.wikipedia.org/wiki/Bias%E2%80%93variance_tradeoff
+   */
+  def getRootVariance(): Option[Seq[Double]] = None
+
+  /**
+   * Get the estimated magnitude of the irreducible noise of each prediction, if possible
+   *
+   * See https://en.wikipedia.org/wiki/Bias%E2%80%93variance_tradeoff
+   */
+  def getIrreducibleError(): Option[Seq[Double]] = None
+
+  /**
+   * Get the "uncertainty", which is the prediction interval if includeNoise and the confidence interval otherwise
+   *
+   * @param includeNoise whether the uncertainty should account for irreducible noise (i.e. a prediction interval)
+   * @return uncertainty of each prediction
+   */
+  override def getUncertainty(includeNoise: Boolean): Option[Seq[Any]] = {
```

Comment:
Thoughts on giving `includeNoise` a default value, in order to limit changes to the interface? My inclination is that we should use a default if it won't be confusing, and that people would generally assume the uncertainty includes irreducible noise.

Reply:
Totally; I meant to do this everywhere (default true) and just missed this one.

Reply:
**`getUncertainty`**

This seems a reasonable interface given the  constraint of backwards compatibility. 

**`RegressionResult`**

* how extensible is this interface with respect to heteroskedastic, and potentially even non-additive noise?
* the descriptions mention "1-sigma", but the distributions are not specified. Is normality assumed here?
* related, the interface crucially relies on intervals; while this is probably fine, it does preclude extensions toward other distributions (for example, multi-modal ones) 
* `getIrreducibleError` is not wrong, but what has motivated this choice over alternatives based on measurement noise, or label noise?

Reply:
`getUncertainty`
Would it be possible to use a 'type' argument instead of the 'includeNoise' boolean? The meaning of includeNoise wouldn't be intuitive to an outside user, while something like 'NormConfIntWidth95' or 'SamplingSDMean' would. This would also allow for extensibility later if we wanted to include other forms of variability measurements. It would also be backwards compatible with a default value of, for example, 'SamplingSDMean' or 'SamplingSDReg'.

`getPredictionInterval` `getConfidenceInterval`
Prediction and confidence interval calculators usually include an argument for setting the level of the interval (e.g. 0.68, 0.95). Allowing this to vary is pretty easy when assuming normality, since you just use the normal or t quantile function to get the right multiplier for the SD. Generally, level=0.68 will not be assumed when you say prediction or confidence interval. Another option would be to stick with 1SD, but rename the methods after the SD being estimated. (This may be tricky for a prediction interval. Off the top of my head, I'm not sure what the base interval width should be called. Prediction error maybe?) 

`getRootVariance`
Why not use getStandardDeviation?

`getIrreducibleError`
Agree that another name might be more interpretable to the wider community. getObservationSD? (Measurement error is also a possibility, but has a different and very specific meaning in stats that we may want to make use of later.

Reply:
> **`RegressionResult`**
> 
> * how extensible is this interface with respect to heteroskedastic, and potentially even non-additive noise?

This assumes heteroskedastic additive noise.  Homoskedastic is a special case of that.  If you wanted to express "I know my noise is homoskedastic" you'd do that in the `Learner` not in the `PredictionResult`

> * the descriptions mention "1-sigma", but the distributions are not specified. Is normality assumed here?
> * related, the interface crucially relies on intervals; while this is probably fine, it does preclude extensions toward other distributions (for example, multi-modal ones)

Returning generic distributions is beyond the scope.  "1-sigma" was meant as a "give me the 2nd moment" in a series-expansion kind of way.

> * `getIrreducibleError` is not wrong, but what has motivated this choice over alternatives based on measurement noise, or label noise?

Alternative choices of names?  Lack of creativity.  I'm very open to suggestions.

Reply:
> `getUncertainty`
> Would it be possible to use a 'type' argument instead of the 'includeNoise' boolean? The meaning of includeNoise wouldn't be intuitive to an outside user, while something like 'NormConfIntWidth95' or 'SamplingSDMean' would. This would also allow for extensibility later if we wanted to include other forms of variability measurements. It would also be backwards compatible with a default value of, for example, 'SamplingSDMean' or 'SamplingSDReg'.
> 

What would be the list of options (and what do they mean)?  Are they sufficiently general to exist in the interface with multiple implementing methods in mind?  I'm not quite sure what `SamplingSDMean` would be, for example.

> `getPredictionInterval` `getConfidenceInterval`
> Prediction and confidence interval calculators usually include an argument for setting the level of the interval (e.g. 0.68, 0.95). Allowing this to vary is pretty easy when assuming normality, since you just use the normal or t quantile function to get the right multiplier for the SD. Generally, level=0.68 will not be assumed when you say prediction or confidence interval. Another option would be to stick with 1SD, but rename the methods after the SD being estimated. (This may be tricky for a prediction interval. Off the top of my head, I'm not sure what the base interval width should be called. Prediction error maybe?)

I like this idea, but note that it contrasts with the standard moment expansion approach that I suggested in my response to @mrupp-citrine .
 
> `getRootVariance`
> Why not use getStandardDeviation?

To distinguish between sqrt(variance) and sqrt(variance + bias^2).  Given the term "bias-variance tradeoff", its not clear to me in what scope "RootVariance" and "StandardDeviation" are synonymous.

> `getIrreducibleError`
> Agree that another name might be more interpretable to the wider community. getObservationSD? (Measurement error is also a possibility, but has a different and very specific meaning in stats that we may want to make use of later.

`getObservationSD` seems reasonable.  What about `getNoise`?